recipes-kernel/linux/files/ti-kernel/0001-iot2050-add-iot2050-platform-support.patch

```diff
@@ -0,0 +1,934 @@
+From 43754bb52f8e7846d96fd48fdbe6d983cb8fc59f Mon Sep 17 00:00:00 2001
+From: Le Jin <le.jin@siemens.com>
+Date: Mon, 18 Nov 2019 17:58:08 +0800
+Subject: [PATCH 1/9] iot2050: add iot2050 platform support
+
+Add support for two iot2050 variants, BASIC and ADVANCED.
+Also add support for fixed gpio number naming.
+
+Signed-off-by: le.jin <le.jin@siemens.com>
+
+Signed-off-by: chao zeng <chao.zeng@siemens.com>
+---
+ arch/arm64/boot/dts/ti/Makefile               |  14 +-
+ arch/arm64/boot/dts/ti/k3-am65-iot2050.dtsi   | 761 ++++++++++++++++++
+ .../boot/dts/ti/k3-am6528-iot2050-basic.dts   |  52 ++
+ .../dts/ti/k3-am6548-iot2050-advanced.dts     |  53 ++
+ 4 files changed, 868 insertions(+), 12 deletions(-)
+ create mode 100644 arch/arm64/boot/dts/ti/k3-am65-iot2050.dtsi
+ create mode 100644 arch/arm64/boot/dts/ti/k3-am6528-iot2050-basic.dts
+ create mode 100644 arch/arm64/boot/dts/ti/k3-am6548-iot2050-advanced.dts
+
+diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
+index 12d72851f94d..d0da438b88a2 100644
+--- a/arch/arm64/boot/dts/ti/Makefile
++++ b/arch/arm64/boot/dts/ti/Makefile
+@@ -8,18 +8,8 @@
+ 
+ DTC_FLAGS += -@
+ 
+-dtb-$(CONFIG_ARCH_K3_AM6_SOC) += k3-am654-base-board.dtb  \
+-				 k3-am654-base-board-sr1.dtb \
+-				 k3-am654-gp.dtbo \
+-				 k3-am654-evm-hdmi.dtbo \
+-				 k3-am654-base-board-jailhouse.dtbo \
+-				 k3-am654-evm-oldi-lcd1evm.dtbo \
+-				 k3-am654-evm-tc358876.dtbo \
+-				 k3-am654-idk.dtbo \
+-				 k3-am654-pcie-usb3.dtbo \
+-				 k3-am654-pcie-usb2.dtbo \
+-				 k3-am654-evm-prupwm.dtbo \
+-				 k3-am654-idk-sr1.dtbo
++dtb-$(CONFIG_ARCH_K3_AM6_SOC) += k3-am6528-iot2050-basic.dtb \
++	k3-am6548-iot2050-advanced.dtb
+ 
+ dtb-$(CONFIG_ARCH_K3_J721E_SOC) += k3-j721e-common-proc-board.dtb \
+ 				   k3-j721e-proc-board-tps65917.dtb \
+diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050.dtsi
+new file mode 100644
+index 000000000000..1380df70b196
+--- /dev/null
++++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050.dtsi
+@@ -0,0 +1,761 @@
++// SPDX-License-Identifier: GPL-2.0
++/*
++ * (C) Copyright 2019 Siemens AG
++ */
++#include <dt-bindings/input/input.h>
++#include <dt-bindings/net/ti-dp83867.h>
++#include <dt-bindings/phy/phy.h>
++
++/ {
++	compatible =  "siemens,iot2050-common", "ti,am654";
++	model = "SIMATIC IOT2050 Common";
```

Comment:
This is meaningless.

Reply:
yes,but this makes it easier for the user to understand its intentions

Reply:
More or less...Anyway we will switch to the upstreaming kernel in the future, so looks fine at this moment.